src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIRequestSmelter.java

```diff
@@ -83,12 +80,37 @@ protected int getActionsDoneUntilDumping()
     }
 
     @Override
-    protected int getExtendedOutputCount(final ItemStack primaryOutput)
+    protected int getExtendedCount(final ItemStack stack)
     {
-        if(currentRecipeStorage != null && currentRecipeStorage.getIntermediate() == Blocks.FURNACE)
+        if (currentRecipeStorage != null && currentRecipeStorage.getIntermediate() == Blocks.FURNACE)
         {
-            return job.getProgress();
+            int count = 0;
+            for (final BlockPos pos : getOwnBuilding().getFurnaces())
+            {
+                if (WorldUtil.isBlockLoaded(world, pos))
+                {
+                    final TileEntity entity = world.getTileEntity(pos);
+                    if (entity instanceof FurnaceTileEntity)
+                    {
+                        final FurnaceTileEntity furnace = (FurnaceTileEntity) entity;
+
+                        final ItemStack smeltableSlot = furnace.getStackInSlot(SMELTABLE_SLOT);
+                        final ItemStack resultSlot = furnace.getStackInSlot(RESULT_SLOT);
+                        if (stack.isItemEqual(smeltableSlot))
```

Comment:
stack.isItemEqual is gonna ignore nbt not sure if thats desired

Reply:
are there nbt smelts?

Reply:
guess can happen with mods

Reply:
Not that I'm aware of, at least in vanilla. I suspect it's possible in modded, and that is likely the way we want to go for the cook's dough.

Reply:
just use our ItemStackUtils instead of vanilla all the time :D